include/boost/beast/core/detail/dynamic_buffer_v0.hpp

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright (c) 2016-2020 Vinnie Falco (vinnie dot falco at gmail dot com)
+// Copyright (c) 2020 Richard Hodges (hodges.r@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_ASIO_CORE_DETAIL_DYNAMIC_BUFFER_V0_HPP
+#define BOOST_ASIO_CORE_DETAIL_DYNAMIC_BUFFER_V0_HPP
+
+#include <boost/asio/buffer.hpp>
+#include <type_traits>
+
+namespace boost {
+namespace beast {
+namespace detail {
+
+template<class Type, class Enable = void>
+struct is_dynamic_buffer_v0 : std::false_type
+{
+};
+
+template<
+    class DynamicBuffer_v0,
+    class MutableBuffersType = typename DynamicBuffer_v0::mutable_buffers_type,
+    class ConstBuffersType = typename DynamicBuffer_v0::const_buffers_type>
```

Comment:
Why are `MutableBuffersType` and `ConstBuffersType` for?

Reply:
I have binned this entire metafunction. It was for detecting and wrapping Asio DynamicBuffer_v1 types. I don't think we need to support them since they're deprecated in ASIO and never previously worked with Beast

Reply:
will revisit if I'm wrong